### PR TITLE
Improve sample configuration file

### PR DESCRIPTION
**Pull request by guilhermelawless**
- Add message at the top of the file with usage information and warning against uncommenting all fields (toml label added because of this change)
- Add warnings for select fields which may impact performance substantially
- Format all fields for consistency
- Add IPC config documentation
- Other small misc improvements

### CODE REVIEW

File: nano/node/node_pow_server_config.cpp [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 nano::error nano::node_pow_server_config::serialize_toml (nano::tomlconfig & toml) const
 {
-	toml.put ("enable", enable, "Enable or disable starting Nano PoW Server as a child process.\ntype:bool");
-	toml.put ("nano_pow_server_path", pow_server_path, "Path to the nano_pow_server executable.\ntype:string,path");
+	toml.put ("enable", enable, "Enable or disable starting Nano PoW Server as a child process\ntype:bool");
+	toml.put ("nano_pow_server_path", pow_server_path, "Path to the nano_pow_server executable\ntype:string,path");
```
Comment (zhyatt) [inline nano/node/node_pow_server_config.cpp]: Did you mean to remove the periods here while newly adding to other locations?

File: nano/node/node_rpc_config.cpp [hunk 2]
```diff
@@ -22,8 +22,8 @@ nano::error nano::node_rpc_config::serialize_toml (nano::tomlconfig & toml) cons
 	toml.put ("enable_sign_hash", enable_sign_hash, "Allow or disallow signing of hashes\ntype:bool");
 
 	nano::tomlconfig child_process_l;
-	child_process_l.put ("enable", child_process.enable, "Enable or disable RPC child process. If false, an in-process RPC server is used.\ntype:bool");
-	child_process_l.put ("rpc_path", child_process.rpc_path, "Path to the nano_rpc executable. Must be set if child process is enabled.\ntype:string,path");
+	child_process_l.put ("enable", child_process.enable, "Enable or disable RPC child process. If false, an in-process RPC server is used\ntype:bool");
```
Comment (zhyatt) [inline nano/node/node_rpc_config.cpp]: More period removals

File: nano/node/nodeconfig.cpp [hunk 3]
```diff
@@ -64,72 +64,72 @@ logging (logging_a)
...
-	toml.put ("vote_generator_threshold", vote_generator_threshold, "Number of bundled hashes required for an additional generator delay\ntype:uint64,[1..11]");
-	toml.put ("unchecked_cutoff_time", unchecked_cutoff_time.count (), "Number of seconds unchecked entry survives before being cleaned\ntype:seconds");
-	toml.put ("tcp_io_timeout", tcp_io_timeout.count (), "Timeout for TCP connect-, read- and write operations\ntype:seconds");
-	toml.put ("pow_sleep_interval", pow_sleep_interval.count (), "The amount to sleep after each batch of POW calculations. Reduces max CPU usage at the expensive of a longer.\ntype:nanoseconds");
+	toml.put ("signature_checker_threads", signature_checker_threads, "Number of additional threads dedicated to signature verification. Defaults to the number of CPU threads minus 1.\ntype:uint64");
+	toml.put ("enable_voting", enable_voting, "Enable or disable voting. Enabling this option requires additional system resources.\ntype:bool");
```
Comment (zhyatt) [inline nano/node/nodeconfig.cpp]: Do we know what additional system resources?

File: nano/node/nodeconfig.cpp [hunk 4]
```diff
@@ -64,72 +64,72 @@ logging (logging_a)
...
+	toml.put ("block_processor_batch_max_time", block_processor_batch_max_time.count (), "The maximum time the block processor can continously process blocks for.\ntype:milliseconds");
+	toml.put ("allow_local_peers", allow_local_peers, "Enable or disable local host peering.\ntype:bool");
+	toml.put ("vote_minimum", vote_minimum.to_string_dec (), "Local representatives do not vote if the delegated weight is under this threshold. Saves on system resources.\ntype:string,amount,raw");
+	toml.put ("vote_generator_delay", vote_generator_delay.count (), "Delay before votes are sent to allow for efficient bundling of hashes in votes.\ntype:milliseconds");
+	toml.put ("vote_generator_threshold", vote_generator_threshold, "Number of bundled hashes required for an additional generator delay.\ntype:uint64,[1..11]");
+	toml.put ("unchecked_cutoff_time", unchecked_cutoff_time.count (), "Number of seconds before deleting an unchecked entry.\nWarning: a low value may result in unsuccessful bootstraps.\ntype:seconds");
```
Comment (zhyatt) [inline nano/node/nodeconfig.cpp]: Any guidance on what a low number might be?

File: nano/node/nodeconfig.cpp [hunk 5]
```diff
@@ -64,72 +64,72 @@ logging (logging_a)
...
+	toml.put ("tcp_io_timeout", tcp_io_timeout.count (), "Timeout for TCP connect-, read- and write operations.\nWarning: a low value may result in TCP connections failing.\ntype:seconds");
```
Comment (zhyatt) [inline nano/node/nodeconfig.cpp]: Any guidance on what a low number might be?

File: nano/node/nodeconfig.cpp [hunk 6]
```diff
@@ -64,72 +64,72 @@ logging (logging_a)
...
-	toml.put ("bandwidth_limit", bandwidth_limit, "Outbound traffic limit in bytes/sec after which messages will be dropped\ntype:uint64");
-	toml.put ("conf_height_processor_batch_min_time", conf_height_processor_batch_min_time.count (), "Minimum write batching time when there are blocks pending confirmation height\ntype:milliseconds");
-	toml.put ("backup_before_upgrade", backup_before_upgrade, "Backup the ledger database before performing upgrades\ntype:bool");
+	toml.put ("confirmation_history_size", confirmation_history_size, "Maximum confirmation history size. If tracking the rate of block confirmations, the websocket feature is recommended instead.\ntype:uint64");
+	toml.put ("active_elections_size", active_elections_size, "Number of active elections. Elections beyond this limit have limited survival time.\nWarning: modifying this value may result in a lower confirmation rate.\ntype:uint64,[250..]");
+	toml.put ("bandwidth_limit", bandwidth_limit, "Outbound traffic limit in bytes/sec after which messages will be dropped.\nNote: changing to unlimited traffic is not recommended.\ntype:uint64");
```
Comment (zhyatt) [inline nano/node/nodeconfig.cpp]: Not sure I wouldn't recommend unlimited traffic. For those who have the bandwidth there shouldn't be an issue in doing so.

File: nano/node/node_pow_server_config.cpp (hunk 1 above)
Comment (guilhermelawless) [reply nano/node/node_pow_server_config.cpp]: Thank you, yes just oversight on these.

File: nano/node/nodeconfig.cpp (hunk 3 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.cpp]: Added: CPU, bandwidth and disk

File: nano/node/nodeconfig.cpp (hunk 4 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.cpp]: After scouting others, anything below 1 second would be very low, playing it safe with 5 seconds for the warning here.

File: nano/node/nodeconfig.cpp (hunk 5 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.cpp]: After scouting others, anything below 1 second would be very low, playing it safe with 5 seconds for the warning here.

File: nano/node/nodeconfig.cpp (hunk 4 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.cpp]: Added 1 hour, mentioning specifically from-scratch bootstraps.

File: nano/node/nodeconfig.cpp (hunk 6 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.cpp]: Added note that the recommendation is for traffic-limited connections.

**Pull request merged**